elasticdl/python/ps/parameters.py

```diff
@@ -0,0 +1,49 @@
+import tensorflow as tf
+
+from elasticdl.python.common.ndarray import tensor_to_ndarray
+from elasticdl.python.ps.embedding_table import create_embedding_table
+
+
+class Parameters(object):
+    def __init__(self):
+        self.init_status = False
+        self.non_embedding_params = {}
+        self.embedding_params = {}
+
+    def get_embedding_param(self, name, indices):
+        if name not in self.embedding_params:
+            raise ValueError(
+                "Please initialize embedding param %s first!", name
+            )
+        return self.embedding_params[name].get(indices)
+
+    def set_embedding_param(self, name, indices, values):
```

Comment:
In my opinion, we could also provide a `set_embedding_param` interface that takes a tensor, so no need to use three arguments and looks cleaner. 

But I'm not sure whether we need to support two interfaces, one takes a tensor, another takes these three arguments. The interface which takes a tensor is more convenient for RPC servicer to call (when sync replica or read from checkpoint file). The interface which takes these three arguments is more convenient for optimizer to call.

Or support these two kinds of arguments in one interface?

Reply:
Since `Tensor` has these three fields, we could reuse the same interface.

```python
a = Tensor()
p = Parameters()
p.set_embedding_param(a.name, a,indices, a.values)
```

If we really need an interface which takes `Tensor` as an argument, we could add it later.

Reply:
By using `Tensor` data structure, we can make the interface cleaner, and this is why I think we nedd a `Tensor` data structure. Otherwise, we add a `Tensor` data structure and change current code from
```python
# current code
# RPC servicer
tensor_pb = get_tensor_pb_from_workers()
name, values, indices = tensor_pb_to_ndarray(tensor_pb)
parameter_instance.set(name, values, indices)
```
to something like
```python
tensor_pb = get_tensor_pb_from_workers()
tensor = tensor_pb_to_tensor(tensor_pb)
parameter_instance_set(tensor.name, tensor.values, tensor.indices)
```

Reply:
Make sense. Since we have `Tensor` data structure, we could use it directly.